src/gateway/upgrade.md

```diff
@@ -51,10 +49,13 @@ Amazon Linux 1 and Debian 8 (Jessie) containers and packages are deprecated and
 #### Blue-green deployments
 
 **Traditional mode**: Blue-green upgrades from versions of 2.8.1 and below to 3.0.0 are not currently supported.
-This is a known issue planned to be fixed in the next 2.8 release. When that version is released, 2.x users should upgrade to that version before beginning a blue-green upgrade to 3.0.
+This is fixed in the 2.8.2 release. 2.x users should upgrade to that version before beginning a blue-green upgrade to 3.0.
 
 **Hybrid mode**: See the [upgrade instructions](#migrate-db) below.
 
+{:.important}
+> **Important**: The legacy hybrid configuration protocol has been removed in favor of [the wRPC protocol introduced in 3.0.0.0](/gateway/changelog/#hybrid-mode). Strictly speaking, this is not a breaking change, but it does have the side effect that rolling upgrades from 2.8.x.y to 3.1.0.0 are not supported. You must upgrade to 3.0.x.y before you can perform a rolling upgrade to 3.1.0.0.  
```

Comment:
I think this notice should live alongside the hybrid mode upgrade instructions. The most current example of this is the `### Upgrade to 3.0.x for hybrid mode` section.

Assuming we are going to continue following the current pattern but with updates for 3.1, here is (loosely) what I would propose:

```markdown
## Upgrade from 1.x/2.x to 3.1.x

### traditional mode

[...]

### hybrid mode

Rolling upgrades from 1.x or 2.x to 3.1.x for hybrid mode are not supported.
To upgrade to 3.1.x, you must first [upgrade to 3.0.x]({{ link to 3.0 upgrade instructions }}).

[...]

## Upgrade from 3.0.x to 3.1.x

### traditional mode

[...]

### hybrid mode

Perform a rolling upgrade of your cluster:

1. Download 3.1.x.
2. Decommission your existing 3.0.x control plane. Your existing 3.0.x data
   planes can continue to handle proxy traffic during this time, even with no 
   active control plane.
4. Configure the new 3.1.x control plane to point to the same data store as
   your old control plane. Run `kong migrations up` and `kong migrations finish`.
5. Start the new 3.1.x control plane.
6. Start new 3.1.x data planes.
7. Gradually divert traffic away from your 3.0.x data planes, and into
   the new 3.1.x data planes. Monitor your traffic to make sure everything
   is going smoothly.
8. When your traffic is fully migrated to the 3.1.x cluster,
   decommission your old 3.0.x data planes.
```

Reply:
@flrgh - based on Lena's good guidance, we do need a new 3.1.x entry for upgrade info and are tracking as a fast follows on https://konghq.atlassian.net/browse/DOCU-2823